sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubsModelFactory.cs

```diff
@@ -143,14 +165,50 @@ public static class EventHubsModelFactory
+        ///   populated with <paramref name="sequenceNumber"/>.
+        /// </remarks>
+        ///
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static EventData EventData(BinaryData eventBody,
+                                          IDictionary<string, object> properties,
+                                          IReadOnlyDictionary<string, object> systemProperties,
+                                          string partitionKey,
+                                          long sequenceNumber,
+                                          long offset,
+                                          DateTimeOffset enqueuedTime) =>
+             new EventData(eventBody, properties, systemProperties, sequenceNumber, string.Empty, enqueuedTime, partitionKey);
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="Azure.Messaging.EventHubs.EventData"/> class.
+        /// </summary>
+        ///
+        /// <param name="eventBody">The data to use as the body of the event.</param>
+        /// <param name="properties">The set of free-form event properties to send with the event.</param>
+        /// <param name="systemProperties">The set of system properties that accompany events read from the Event Hubs service.</param>
+        /// <param name="partitionKey">The partition hashing key associated with the event when it was published.</param>
+        /// <param name="sequenceNumber">The sequence number assigned to the event when it was enqueued in the associated Event Hub partition.</param>
+        /// <param name="globalOffset">The global offset of the event when it was received from the associated Event Hub partition.</param>
+        /// <param name="offset">The offset of the event when it was received from the associated Event Hub partition.</param>
+        /// <param name="enqueuedTime">The date and time, in UTC, of when the event was enqueued in the Event Hub partition.</param>
+        ///
+        /// <remarks>
+        ///   The Event Hubs service no longer uses offsets with numeric values. The <paramref name="offset"/> parameter is
+        ///   included for backwards compatibility, but is ignored. Instead, <see cref="EventData.Offset"/> is
+        ///   populated with <paramref name="sequenceNumber"/>.
+        /// </remarks>
+        ///
         public static EventData EventData(BinaryData eventBody,
                                           IDictionary<string, object> properties = null,
                                           IReadOnlyDictionary<string, object> systemProperties = null,
                                           string partitionKey = null,
                                           long sequenceNumber = long.MinValue,
+                                          string globalOffset = null,
                                           long offset = long.MinValue,
```

Comment:
Ok actually I was going to call this one to see if you had ideas. When I made this overload without offset, then if a customer does something like:
`EventHubsModelFactory.EventData(binaryData)` it gets an ambiguous reference error.

The only way to fix it is to make all the previous arguments not optional to differentiate between the two. But then not having offset here would break this:
`EventHubsModelFactory.EventData(binaryData, offset: 1)`
Which seems bad.

Reply:
Ok that sounds like a good place. I updated this to remove offset for now.